software/spmd/bsg_barrier/Makefile

```diff
@@ -8,7 +8,7 @@ include ../Makefile.include
 
 all: main.run
 
-main.riscv:  $(OBJECT_FILES) $(SPMD_COMMON_OBJECTS) $(BSG_MANYCORE_LIB) ../common/crt.o
+main.riscv: $(LINK_SCRIPT)  $(OBJECT_FILES) $(SPMD_COMMON_OBJECTS) $(BSG_MANYCORE_LIB) ../common/crt.o
 	$(RISCV_LINK) $(OBJECT_FILES) $(SPMD_COMMON_OBJECTS) -L. "-l:$(BSG_MANYCORE_LIB)" -o $@ $(RISCV_LINK_OPTS)
```

Comment:
Not for this PR, but a thought:

Do MOST of the SPMD programs have the same compilation rule? The answer here is obviously no, because software/spmd/branch/Makefile has a different rule. However, many look similar. Is there an important reason why they differ, when they differ? Could we move the main.riscv compilation rule so we don't have to edit 100-ish files every time we do a change like this? Perhaps to tail_rules?

Just a thought

Reply:
For example, we  have discussed the benefit of compiling compiling crt.o in each directory. instead of using a common path. This is another PR that would modify 100 ish files in very similar ways

Reply:
Oh missed this! May be we could factor out common prerequisites and common argument `$(RISCV_LINK)` command?

In this particular case though, it was a minor bug to not have `$(LINK_SCRIPT)` one of the prerequisites, I think.